### DISCUSSION TITLE
Angular wrapper

### DISCUSSION THREAD

**Discussion by NaNgets** [mathuo/dockview#413] [Ideas]
Hi,

Would you consider changing the name of `dockview` to `dockview-react` and add a `dockview-angular` wrapper?

I'm able to write the angular wrapper if needed :)

Thanks.

**Comment by NaNgets** [comment]
@mathuo is this something you can consider? Thanks :)

**Comment by techbech** [comment]
Changing the name is a major change and will break backward compatibility.
So let me suggest that you make the Angular wrapper first, and then the strategy for naming can be considered afterwards.

**Comment by mathuo** [comment]
If dockview did start supporting multiple frameworks it would make sense to rename dockview-core to dockview and then maintain packages appropiately named dockview-<framework> (e.g. dockview-react, dockview-angular etc.)

It would however be a dramatic break in functionality so to start with at least it probably wouldn't follow those names but I do think it would be worth the breaking change once dockview has at least one additional framework supported and we could achieve that through a major version release (e.g. 1.x.x -> 2.x.x)

Short answer is yes the idea is for dockview to support multiple frameworks through seperate packages but having ideal namings may take a little time.

**Comment by NaNgets** [comment]
I think it's much simpler - 

- Keep dockview-core as the core library,
- Convert dockview to dockview-react (i.e. discontinue dockview)
- Add a new package named dockview-angular

That way you're not breaking the change of the dockview package, projects will just need to use the new name and not worry about it suddenly changing entirely.

But, as suggested, I will first build the wrapper as suggested.

**Reply by imsujeet** [reply]
Hi @NaNgets , 
Were you able to create the Angular wrapper?
If yes, it would be great if you can help building me one.

**Comment by martinzima** [comment]
Sorry for reviving an old thread, but as I saw that the Angular wrapper was still not implemented, I created a proof-of-concept wrapper myself.

https://github.com/martinzima/ngx-dockview

Compared to the React wrapper, I chose a slightly more declarative approach (over imperative API) that I felt was more natural for Angular library.

@mathuo Would you be willing to review this and possibly, if I created a PR, merge it into the core project? (Once we made sure it was well aligned with the rest of the Dockview libraries.)
